src/metabase/api/action.clj

```diff
@@ -89,15 +89,13 @@
    template               [:maybe http-action-template]
    response_handle        [:maybe json-query-schema]
    error_handle           [:maybe json-query-schema]}
-
-  (api/write-check Card model_id)
   (when (and (nil? database_id)
              (= "query" type))
     (throw (ex-info (tru "Must provide a database_id for query actions")
                     {:type        type
                      :status-code 400})))
-  (when database_id
-    (actions/check-actions-enabled! (db/select-one Database :id database_id)))
+  (let [model (api/write-check Card model_id)]
```

Comment:
any reason why this is better than having `(api/write-check Card model_id)` at top?

IMO write check should be the 1st thing to do.

Reply:
I thought schema validation is generally first. And the check for a non-nil database_id is part of that